template/template.go

```diff
@@ -0,0 +1,205 @@
+package template
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+
+	ast "go.expect.digital/mf2/parse"
+)
+
+// MessageFormat2 Errors as defined in the specification.
+//
+// https://github.com/unicode-org/message-format-wg/blob/122e64c2482b54b6eff4563120915e0f86de8e4d/spec/errors.md
+var (
+	ErrSyntax                = errors.New("syntax error")
+	ErrUnresolvedVariable    = errors.New("unresolved variable")
+	ErrUnknownFunction       = errors.New("unknown function reference")
+	ErrDuplicateOptionName   = errors.New("duplicate option name")
+	ErrUnsupportedExpression = errors.New("unsupported expression")
+	ErrFormatting            = errors.New("formatting error")
+)
+
+type ExecFn func(operand any, opts map[string]any) (string, error)
+
+type Template struct {
+	ast       *ast.AST
+	execFuncs map[string]ExecFn
```

Comment:
?

```suggestion
	funcs map[string]Func
```

Reply:
i was going to similarities with template pkg
https://cs.opensource.google/go/go/+/refs/tags/go1.22.0:src/text/template/template.go;l=23